airbyte-cdk/python/airbyte_cdk/sources/configurable/requesters/paginators/offset_pagination.py

```diff
@@ -0,0 +1,30 @@
+#
+# Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+#
+from typing import Any, List, Mapping, Optional
+
+import requests
+from airbyte_cdk.sources.configurable.requesters.paginators.paginator import Paginator
+from airbyte_cdk.sources.configurable.states.dict_state import DictState
+
+
+class OffsetPagination(Paginator):
+    def __init__(self, page_size: int, state: DictState, tag: str = "offset"):
+        self._limit = page_size
+        self._state: DictState = state
+        self._tag = tag
+        self._update_state_with_offset(0)
+
+    def next_page_token(self, response: requests.Response, last_records: List[Mapping[str, Any]]) -> Optional[Mapping[str, Any]]:
+        if len(last_records) < self._limit:
+            return None
+        offset = self._get_offset() + self._limit
+        token_map = {self._tag: offset}
+        self._update_state_with_offset(offset)
+        return token_map
+
+    def _update_state_with_offset(self, offset):
+        self._state.update_state(**{self._tag: offset})
```

Comment:
offset seems like a transient variable, why store it in the state instead of as an instance var? is it supposed to be used across invocations of the connector?

Reply:
@sherifnada the idea is to keep all state in the State object so everything else can be stateless. the `stream_state`, and `stream_slice` then become 2 out of possibly many values that can be tracked during a sync